### PR TITLE
[maint] changelists for noetic 1.1.0

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

Here are the changelists for noetic.

One thing to note is that I brought over changelist changes from `melodic-devel` and fixed up the output of the changelist generator.

### CODE REVIEW

**Comment by rhaschke** [comment]
We cannot yet release into Noetic. There is a major road blocker: https://github.com/ros/geometry2/pull/471.

**Comment by tylerjw** [comment]
> We cannot yet release into Noetic. There is a major road blocker: [ros/geometry2#471](https://github.com/ros/geometry2/pull/471).

Thank you for pointing this out.  I had time to work on this today so I wanted to do as much as I could to get closer towards it.  Doing the changelogs is kind of annoying and I figured I could get a bunch of that done ahead of time.

File: moveit_plugins/moveit_simple_controller_manager/CHANGELOG.rst
Comment (tylerjw) [inline moveit_plugins/moveit_simple_controller_manager/CHANGELOG.rst:19]: remove this line

**Comment by rhaschke** [comment]
Dear @tylerjw, we should wait for the next Noetic sync, which is [scheduled for the weekend](https://discourse.ros.org/t/preparing-for-noetic-sync-2020-08-29/16125).

**Comment by tylerjw** [comment]
If we wait it'll be a month or so before it is actually released then.  Seeing as this is sort of an "unstable" release I don't see any harm in releasing today.

**Comment by tylerjw** [comment]
I pushed an update of this.  Please review for correctness.  @rhaschke @v4hn 

As to when we release, Dave would like to release sooner rather than later.  I don't see much harm in doing that as we already agreed there is a period of 6mo or so where we can release again and break the API.

One thing I'm sort of confused about is what do we do about releases made off of master (the ones on melodic-devel) and their changelists.  I manually brought over those changelists and their versions into these files to keep them up to date.  However that isn't really what happened on master (as the history of master and melodic-devel diverged with some changes being backported and releases happening on those subsets of changes on melodic-devel).

**Comment by davetcoleman** [comment]
I don't see the harm in making the release to rosdistro now and they can decide if they let it into this sync or not. It still gets our release out.  @tylerjw I think you should proceed

I just requested they wait though, and it might be convincing as MoveIt is a big deal in the ROS ecosystem 
https://discourse.ros.org/t/preparing-for-noetic-sync-2020-08-29/16125/2

File: moveit_planners/moveit_planners/CHANGELOG.rst
Comment (davetcoleman) [inline moveit_planners/moveit_planners/CHANGELOG.rst:4]: Shouldn't there be a 1.1.0  here?

File: moveit_planners/ompl/CHANGELOG.rst
Comment (davetcoleman) [inline moveit_planners/ompl/CHANGELOG.rst:4]: Shouldn't there be a 1.1.0 here?

File: moveit_planners/ompl/CHANGELOG.rst
Comment (davetcoleman) [inline moveit_planners/ompl/CHANGELOG.rst:149]: why did this get re-instated?

File: moveit_planners/ompl/CHANGELOG.rst
Comment (tylerjw) [reply moveit_planners/ompl/CHANGELOG.rst:4]: fixed

File: moveit_planners/moveit_planners/CHANGELOG.rst
Comment (tylerjw) [reply moveit_planners/moveit_planners/CHANGELOG.rst:4]: fixed

File: moveit_planners/ompl/CHANGELOG.rst
Comment (tylerjw) [reply moveit_planners/ompl/CHANGELOG.rst]: idk... put it back

**Comment by rhaschke** [comment]
> I don't see the harm in making the release to rosdistro now and they can decide if they let it into this sync or not.

@sloretz is holding back new releases anyway. If we just wait for the sync, we can fix/cleanup our Docker images before merging noetic-devel (#2178). Currently rosdep cannot (yet) fetch the required packages in the `noetic-ci` image, because they are available in `testing` only.

> I just requested they wait though, and it might be convincing as MoveIt is a big deal in the ROS ecosystem

Pushing a late release just before the sync deadline is defeating the idea of soak time. I'm not supporting this at all.
We are talking about a few days of delay, not a month!

**Comment by tylerjw** [comment]
The release happened so this should be ready to go.

**Comment by tylerjw** [comment]
Ok, this is finally ready for review / merge.  I found several places where the last time I ran the generate changelog command didn't work and fixed it up.  I believe we are now ready for a noetic release.

**Comment by tylerjw** [comment]
CI is failing for kinetic in geometric_shapes:

```bash
Errors     << geometric_shapes:cmake /root/ros_ws/logs/geometric_shapes/build.cmake.000.log
CMake Error at /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find Qhull (missing: QHULL_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindQhull.cmake:48 (find_package_handle_standard_args)
  CMakeLists.txt:58 (find_package)
```

**Review by davetcoleman — APPROVED** [review]

**Comment by davetcoleman** [comment]
> If we can't build geometric_shapes in kinetic, should we stop testing kinetic in CI?

We decided a while back during a Maintainer Meeting to drop support for Kinetic in the master branch, so yes, let's stop testing Kinetic. That's far too many distributions to attempt to support with our limited resources (16.04, 18.04, 20.04, and ROS 2 on 20.04)

**Comment by tylerjw** [comment]
Dropping kinetic from CI was merged, rebased to include that.

**Review by rhaschke — COMMENTED** [review]
You said you migrated changelogs from Melodic. If so, wouldn't it be more reasonable to list those changes below corresponding tags 1.0.x and only focus on really new contributions (compared to the last melodic release 1.0.6) below 1.1.0? Now, we have a looong list of changes below 1.1.0.

File: moveit_planners/ompl/CHANGELOG.rst
Comment (tylerjw) [inline moveit_planners/ompl/CHANGELOG.rst:35]: @rhaschke here is an example of how it worked how I thought it would.  Looking through some of them it didn't work.  I'll fix that tomorrow morning.  Good catch.

Comment (rhaschke) [reply moveit_planners/ompl/CHANGELOG.rst:35]: That what I had in my mind. Thanks for reworking this!

Comment (tylerjw) [reply moveit_planners/ompl/CHANGELOG.rst:35]: > You said you migrated changelogs from Melodic. If so, wouldn't it be more reasonable to list those changes below corresponding tags 1.0.x and only focus on really new contributions (compared to the last melodic release 1.0.6) below 1.1.0? Now, we have a looong list of changes below 1.1.0.

See my comment.  Looking through the changes I did think some of them were way longer than I expected.  Also, because the history of master and melodic-devel diverged (some old changes are only on master, and some newer ones are on melodic-devel) that makes this a bit more challenging, but I'll get it done.

Comment (tylerjw) [reply moveit_planners/ompl/CHANGELOG.rst:35]: I wrote a script to do this... the short answer is it didn't work completely and I didn't do a good enough job of checking the results... I'll fix this up.

**Comment by tylerjw** [comment]
I fixed this up.  All changelog files now have the changes ported from the melodic-devel branch to show only the changes in this release not in the last melodic release.

One thing to consider is that this require constant fixups in the future if we port some of the latest changes into melodic-devel.  I'm not sure if this is the right approach.  It might have been better to have the huge changelist here that is everything since 1.0.1 instead of from 1.0.6.  As the histories will have diverged if we choose what I did here I think we should consider this the truth for master (even if some things are later backported) and not copy in any more changelogs from melodic-devel.

**Comment by tylerjw** [comment]
As if part of some sort of cruel joke tests are now failing on Noetic:

```
build/moveit_core/test_results/moveit_core/gtest-test_aabb.xml: 6 tests, 0 errors, 4 failures, 0 skipped
build/moveit_core/test_results/moveit_core/gtest-test_bullet_continuous_collision_checking.xml: 8 tests, 0 errors, 0 failures, 2 skipped
build/moveit_core/test_results/moveit_core/gtest-test_robot_model.xml: 6 tests, 0 errors, 2 failures, 0 skipped
build/moveit_core/test_results/moveit_core/gtest-test_robot_state.xml: 8 tests, 0 errors, 4 failures, 0 skipped
Summary: 540 tests, 0 errors, 10 failures, 2 skipped
```

I don't know why, but this should probably be fixed before we can release Noetic.

(˚Õ˚)ر ~~~~╚╩╩╝

**Comment by tylerjw** [comment]
Created issue for the failing tests here: https://github.com/ros-planning/moveit/issues/2298

**Review by rhaschke — COMMENTED** [review]
Thanks for this great preparation, @tylerjw. I just skimmed through the changelogs and cleaned some simple things. Could you please rebase onto noetic-devel?

**Review by rhaschke — APPROVED** [review]

**Comment by tylerjw** [comment]
Do we want to release off of noetic-devel or master?  @rhaschke I thought we were going to release off of master for Noetic.

**Comment by rhaschke** [comment]
> Do we want to release off of noetic-devel or master? I thought we were going to release off of master for Noetic.

We should release from `noetic-devel`. Until the end of the year, we should keep `noetic-devel` and `master` in sync. Afterwards, we will branch off `master` to include incompatible changes only.

**Comment by rhaschke** [comment]
I'm going to prepare the release now.

**Comment by rhaschke** [comment]
Release PR: https://github.com/ros/rosdistro/pull/26484

**Comment by rhaschke** [comment]
@tylerjw: There are two issues with the current release:
On Ubuntu Focal, the [`subframes_test` was failing](http://build.ros.org/job/Ndev__moveit__ubuntu_focal_amd64/1). Was this flaky in the past?
On Debian Buster, [gcc fails to find a system include](http://build.ros.org/job/Ndev_db__moveit__debian_buster_amd64/1)?!?

**Comment by rhaschke** [comment]
I shortly looked into this today.
The first issue (subframes_test failing), is due panda_moveit_config being used. I filed a fix: #2300.
The second issue is more complicated. See https://github.com/ros/rosdistro/pull/26484#issuecomment-688687613 for an analysis.

**Pull request merged**
